environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -83,24 +84,32 @@ public EnvironmentCreationService(
     }
 
     public EnvironmentDto create(EnvironmentCreationDto creationDto) {
-        if (environmentService.isNameOccupied(creationDto.getName(), creationDto.getAccountId())) {
-            throw new BadRequestException(String.format("Environment with name '%s' already exists in account '%s'.",
-                    creationDto.getName(), creationDto.getAccountId()));
+        Environment environment = new Environment();
+        try {
```

Comment:
Would be nicer if you moved the beginning of the try { after the validateCreation call and then creating this empty env variable would be unnecessary.

Reply:
Also, in this case the environment is being saved even when some validation error happened and we should just reject the request with a http 400.